src/components/charts/lineChart/LineChart.tsx

```diff
@@ -0,0 +1,167 @@
+import { scaleLinear, scaleTime, line, timeFormat, max, Line, ScaleLinear, ScaleTime } from 'd3'
+import { useTranslation } from 'react-i18next'
+import { v4 as uuid } from 'uuid'
+
+interface IService {
+  /**
+   * name of the service
+   */
+  name: string
+  /**
+   * status of the service
+   */
+  status: string
+  /**
+   * response time for the service
+   */
+  response_time: number
+  /**
+   * check time for the service
+   */
+  check_time: string
+}
+
+interface IProps {
+  data: Array<IService>
+}
+
+// margin convention often used with D3
+const margin = { top: 20, right: 30, bottom: 40, left: 60 }
+const width = 700 - margin.left - margin.right
+const height = 350 - margin.top - margin.bottom
+
+const color: { [key: string]: string } = {
+  OK: '#17324D',
+  'n/d': '#A2ADB8',
+  KO: '#FE6666',
+}
+
+/**
+ * Line chart component
+ * @component
+ */
+export const LineChart: React.FC<IProps> = ({ data }) => {
+  const { t } = useTranslation(['detailsPage'])
+
+  // header of the chart
+  const header: JSX.Element = (
+    <g className="bar-header" transform={`translate(${margin.left}, ${margin.top})`}>
+      <text>
+        <tspan fontFamily="Titillium Web" fontSize="18px" color="#17324D" fontWeight="700">
+          {t('lineChartTitle')}
+        </tspan>
+      </text>
+    </g>
+  )
+
+  //x scale
+  const x: ScaleTime<number, number, never> = scaleTime()
+    .range([0, width])
+    .domain([
+      // get the previous date of the first check_date in the array
+      // because the chart is not including the first one
+      new Date(new Date(data[0].check_time).setDate(new Date(data[0].check_time).getDate() - 1)),
+      new Date(data[data.length - 1].check_time),
+    ])
+
+  //y scale
+  const y: ScaleLinear<number, number, never> = scaleLinear()
+    .domain([
+      0,
+      max(data, (d: IService) => {
+        return d.response_time || null
+      }),
+    ] as Array<number>)
+
+    .range([height, 0])
+
+  // create the x ticks and create from them the vertical lines of the grid
+  const xTicks: JSX.Element = (
+    <g
+      transform={`translate(0, ${height})`}
+      fontFamily="Poppins"
+      opacity={0.4}
+      fontSize={10}
+      textAnchor="middle"
+    >
+      {x.ticks().map((d: Date) => (
+        <g opacity="1" className="tick" transform={`translate(${x(d)}, 0)`} key={uuid()}>
```

Comment:
Il valore del prop key deve essere stabile e non rigenerato ad ogni render. 
Più info [qui](https://react.dev/learn/rendering-lists#rules-of-keys).


Non conosco bene il contesto, ma per questo specifico caso utilizzerei `d.getTime()`.